open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/manage/validator/VRPConsentRequestValidator.java

```diff
@@ -0,0 +1,608 @@
+
+            Object debtorAccount = initiation.get(ConsentExtensionConstants.DEBTOR_ACC);
+
+            if (!isValidJSONObject(debtorAccount)) {
+                return ConsentManageUtil.getValidationResponse(ErrorConstants.PAYLOAD_FORMAT_ERROR_DEBTOR_ACC,
+                        ErrorConstants.INVALID_PARAMETER_DEBTOR_ACC,
+                        ErrorConstants.PATH_DEBTOR_ACCOUNT);
+            }
+
+            JSONObject validationResult = ConsentManageUtil.validateDebtorAccount((JSONObject) debtorAccount);
+
+            if (!(Boolean.parseBoolean(validationResult.getAsString(ConsentExtensionConstants.IS_VALID)))) {
+                log.error(validationResult.get(ConsentExtensionConstants.ERRORS));
+                return validationResult;
+            }
+
+        } else {
+            log.error(ErrorConstants.PAYLOAD_FORMAT_ERROR_DEBTOR_ACC);
+            return ConsentManageUtil.getValidationResponse(ErrorConstants.RESOURCE_INVALID_FORMAT,
+                    ErrorConstants.PAYLOAD_FORMAT_ERROR_DEBTOR_ACC, ErrorConstants.PATH_REQUEST_BODY);
+        }
+
+        //Validate CreditorAccount
+        if (initiation.containsKey(ConsentExtensionConstants.CREDITOR_ACC)) {
+
+            Object creditorAccount = initiation.get(ConsentExtensionConstants.CREDITOR_ACC);
+
+            if (!isValidJSONObject(creditorAccount)) {
+                return ConsentManageUtil.getValidationResponse(ErrorConstants.PAYLOAD_FORMAT_ERROR_CREDITOR_ACC,
+                        ErrorConstants.INVALID_PARAMETER_CREDITOR_ACC,
+                        ErrorConstants.PATH_CREDIT_ACCOUNT);
+            }
+
+            JSONObject validationResult = ConsentManageUtil.validateCreditorAccount((JSONObject) creditorAccount);
+
+            if (!(Boolean.parseBoolean(validationResult.getAsString(ConsentExtensionConstants.IS_VALID)))) {
+                log.error(validationResult.get(ConsentExtensionConstants.ERRORS));
+                return validationResult;
+            }
+
+        } else {
+            log.error(ErrorConstants.PAYLOAD_FORMAT_ERROR_CREDITOR_ACC);
+            return ConsentManageUtil.getValidationResponse(ErrorConstants.RESOURCE_INVALID_FORMAT,
+                    ErrorConstants.PAYLOAD_FORMAT_ERROR, ErrorConstants.PATH_REQUEST_BODY);
+        }
+
+        validationResponse.put(ConsentExtensionConstants.IS_VALID, true);
+        return validationResponse;
+    }
+
+    /**
+     * Validates the presence of a specified key in a JSONObject (either the amount or the currency)
+     * and checks if the associated value is a non-empty string.
+     *
+     * @param parentObj The JSONObject to be validated.
+     * @param key       The key to be checked for presence in the parentObj.
+     * @return true if the specified key is present in the parentObj and the associated value is a
+     * non-empty string.
+     */
+    public static boolean validateAmountCurrency(JSONObject parentObj, String key) {
```

Comment:
Since this method validates json object's given key is present and non empty, shall we give a generalise method name like `validateJsonObjectKey` or something like that? WDYT?

Reply:
Fixed in [5ba1c85]